python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -321,6 +307,9 @@ def on_opened(self, opened_payload: PipesOpenedData) -> None:
     def on_launched(self, launched_payload: PipesLaunchedData) -> None:
         self.launched_payload = launched_payload
 
+    @abstractmethod
+    def can_start(self, params: PipesParams) -> bool: ...
```

Comment:
We need a better name than `can_start` since it's not obvious what "start" means (e.g. the threads are already running when we are checking this). Maybe `are_messages_readable`.